### ISSUE TITLE
Permission dialog re-opened again after app shutdown

### ISSUE THREAD

**Issue by tecnonucleo** [ParkSangGwon/TedPermission#94]
Hello. I had an issue trying to end the app when some required dialog is denied.
In my Application object, I have this code (in Kotlin):

```kotlin
val objPermissionListener = object : PermissionListener {
 override fun onPermissionGranted() {
  onItfCreate_2(pIfrForm)
 }

 override fun onPermissionDenied(deniedPermissions: List<String>) {
  terminateApplication()
 }
}

TedPermission.with(pIfrForm)
	.setRationaleTitle("Rational title")
	.setRationaleMessage("Rational message")
	.setPermissionListener(objPermissionListener)
	.setDeniedMessage("If you reject permission,you can not use this service\n\nPlease turn on permissions at [Setting] > [Permission]")
	.setPermissions(Manifest.permission.READ_PHONE_STATE, Manifest.permission.WRITE_EXTERNAL_STORAGE)
	.check()

/**
 * Perform a special termination when the application is not completely initialized yet.
 * It's used to terminate the application when some of the required permissions are denied. In this special
 * case it's necessary to close the app because leaving it partially initialized leads to several errors.
 */
private fun terminateApplication() {
    mIfrCurrentActivity?.finish()
    System.exit(-1)
}
```

I downloaded the source code and deployed it inside of my project. Could fix the problem by applying a little change in the method TedPermissionsActivity.permissionResult()
Basically, I moved the Activity finishing code so it’s executed BEFORE the call to the registered listeners. Its’ because when System.exit() is executed, if there is an Activity that remains opened, the application restarts rather than finishing definitely. And the Activity that remained opened was TedPermissionsActivity.
Original code:

```java
private void permissionResult(List<String> deniedPermissions) {
    Log.v(TedPermission.TAG, "permissionResult(): " + deniedPermissions);
    if (permissionListenerStack != null) {
        PermissionListener listener = permissionListenerStack.pop();

        if (ObjectUtils.isEmpty(deniedPermissions)) {
            listener.onPermissionGranted();
        } else {
            listener.onPermissionDenied(deniedPermissions);
        }
        if (permissionListenerStack.size() == 0) {
            permissionListenerStack = null;
        }
    }

    finish();
    overridePendingTransition(0, 0);
}
```

Modifyed code:

```java
private void permissionResult(List<String> deniedPermissions) {
    Log.v(TedPermission.TAG, "permissionResult(): " + deniedPermissions);

    finish();
    overridePendingTransition(0, 0);

    if (permissionListenerStack != null) {
        PermissionListener listener = permissionListenerStack.pop();

        if (ObjectUtils.isEmpty(deniedPermissions)) {
            listener.onPermissionGranted();
        } else {
            listener.onPermissionDenied(deniedPermissions);
        }
        if (permissionListenerStack.size() == 0) {
            permissionListenerStack = null;
        }
    }
}
```
Maybe you can apply this change to the master branch…
Regards, German Blando

**Comment by ParkSangGwon** [comment]
@tecnonucleo Thanks. This is solved at v2.2.2

**Comment by tecnonucleo** [comment]
Glad to know that!
